server/lyft/decorators/audit_project_commands_wrapper.go

```diff
@@ -0,0 +1,148 @@
+package decorators
+
+import (
+	"encoding/json"
+	"strconv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/metrics"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+)
+
+// AuditProjectCommandWrapper is a decorator that notifies sns topic
+// about the state of the command. It is used for auditing purposes
+type AuditProjectCommandWrapper struct {
+	SnsWriter sns.Writer
+	events.ProjectCommandRunner
+}
+
+func (p *AuditProjectCommandWrapper) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
+	ctx.SetScope("audit_applies")
+
+	id := uuid.New()
+	startTime := strconv.FormatInt(time.Now().Unix(), 10)
+
+	applyEvent := &ApplyEvent{
+		Version:        1,
+		ID:             id.String(),
+		RootName:       ctx.ProjectName,
+		JobType:        models.ApplyCommand,
+		Respository:    ctx.BaseRepo.FullName,
+		Environment:    ctx.Tags["environment"],
+		PullNumber:     ctx.Pull.Num,
+		InitiatingUser: ctx.User.Username,
+		Project:        ctx.Tags["service_name"],
+		ForceApply:     ctx.ForceApply,
+		StartTime:      startTime,
+		Revision:       ctx.Pull.HeadCommit,
+	}
+
+	if err := p.emit(ctx, ApplyEventRunning, applyEvent); err != nil {
```

Comment:
One thing I realized is that on the deploys side, they fail a job when they can't track it.  This makes sense from a compliance/auditing standpoint I think, so maybe we need to modify this logic to do so.